pkg/cluster/update.go

```diff
@@ -248,34 +242,36 @@ func ssNameForVM(vm *compute.VirtualMachineScaleSetVM) scalesetName {
 }
 
 // updateInPlace updates one by one all the VMs of a scale set, in place.
-func (u *simpleUpgrader) updateInPlace(ctx context.Context, cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole, ssHashes map[scalesetName]hash) *api.PluginError {
-	vms, err := u.listVMs(ctx, cs, role)
+func (u *simpleUpgrader) updateInPlace(ctx context.Context, cs *api.OpenShiftManagedCluster, app *api.AgentPoolProfile, ssHashes map[scalesetName]hash) *api.PluginError {
+	vms, err := u.listVMs(ctx, cs, app)
 	if err != nil {
 		return &api.PluginError{Err: err, Step: api.PluginStepUpdateInPlaceListVMs}
 	}
 
-	sorted, err := u.sortMasterVMsByHealth(vms, cs)
-	if err != nil {
-		return &api.PluginError{Err: err, Step: api.PluginStepUpdateInPlaceSortMasters}
-	}
-
 	blob, err := u.readUpdateBlob()
 	if err != nil {
 		return &api.PluginError{Err: err, Step: api.PluginStepUpdateInPlaceReadBlob}
 	}
 
-	sorted = u.filterOldVMs(sorted, blob, ssHashes)
-	for _, vm := range sorted {
+	// range our vms in order, so that if we previously crashed half-way through
+	// updating one and it is broken, we pick up where we left off.
+	sort.Slice(vms, func(i, j int) bool {
```

Comment:
Isn't this sort mute since `updateInPlace` runs in `sortedAgentPoolProfilesForRole`?

Reply:
Sorry, I misread `sortedAgentPoolProfilesForRole`, nevermind.